code/workspaces/web-app/public/storage_config.json

```diff
@@ -0,0 +1,26 @@
+{
```

Comment:
I don't think so, and it's following the same pattern as web_auth_config.json.
* The 'master' version is contained in the configmap, https://github.com/NERC-CEH/datalab-k8s-manifests/blob/NERCDL-679-config-map-for-storage/templates/datalab/storage-configmap.template.yml
* You need a version for local development, to be used in docker-compose - that's in code/development-env/config/local/storage_config.json.
* In order for the docker-compose volumes to work, the files need to exist in their expected locations.  I could have put empty files in, but for unit tests it made sense to populate them.  These files have an extra line, 
```
  "comment": "THIS FILE IS FOR TEST PURPOSES ONLY",
```
* The infrastructure and client workspaces both use the version in workspaces/common.
* The web_app needs to be in public, so it can be volume mounted after the webapp is built.
The only way to reduce the number of versions would be for the infrastructure and client workspaces to depend on the one in web-app/public, which feels wrong.

Reply:
That all makes sense, thanks for clarifying Andy!